src/libvirt-dbus.js

```diff
@@ -1462,6 +1469,28 @@ function initResource(connectionName, method, updateOrAddMethod, flags) {
             .catch(ex => console.warn('initResource action failed:', ex.toString()));
 }
 
+export function migrateToUri(connectionName, objPath, destUri, storage, live, temporary) {
+    // direct migration is not supported by QEMU, so it's opposite, the P2P migration should always be used
+    let flags = Enum.VIR_MIGRATE_PEER2PEER;
+
+    if (!live) {
+        flags = flags | Enum.VIR_MIGRATE_OFFLINE | Enum.VIR_MIGRATE_PERSIST_DEST;
+    } else {
+        flags = flags | Enum.VIR_MIGRATE_LIVE;
+
+        if (!temporary)
+            flags = flags | Enum.VIR_MIGRATE_PERSIST_DEST;
+    }
+
+    if (storage === "copy")
+        flags = flags | Enum.VIR_MIGRATE_NON_SHARED_DISK;
```

Comment:
Sure both are valid. Copying from   https://libvirt.org/html/libvirt-libvirt-domain.html#virDomainMigrateFlags: 
```
The VIR_MIGRATE_PARAM_MIGRATE_DISKS parameter can be used to specify which disks should be migrated.
```

So in one case you tell libvirt 'copy my disk' in the other case migration is storage unaware, and the user needs to have their disks on a shared storage between the source and destination. Or just somehow like our tests do have duplicated the disk in the destination, which actually could make sense for offline migration. (Destination VM is shut off)

Reply:
"By default only non-shared non-readonly disk images are transferred." -- so that means it should always copy the VM's "primary" disk with the root file system. So this kind of *is* an "unbreak my setup" option -- if the extra disks are on shared storage, the user should uncheck the storage box; and if they are not shared with the destination already, then the user should check the box?

Reply:
But the "primary" disk can also be on shared storage. If it is on the shared storage, user doesn't want to use VIR_MIGRATE_NON_SHARED_DISK. If it's a local disk with the root file system, user wants to use VIR_MIGRATE_NON_SHARED_DISK.
So I don't it's a "unbreak my setup" option.